File: CFPractice/A_Remove_Smallest.cpp
#include <bits/stdc++.h>
#include <sstream>
#include <algorithm>
#define ll long long
#define MAXN 100001
#define mod 1000000007
using namespace std;
#define pb push_back

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    ll t;
    cin >> t;
    while (t--)
    {
        ll n, cnt;
        vector<int> arr;
        cin >> n;
        for (int i = 0; i < n; i++)
        {
            ll x;
            cin >> x;
            arr.pb(x);
        }

        sort(arr.begin(), arr.end());
        for (int i = 0; i < n; i++){
            for (int j = 0; j < n; j++){
                if(i!=j && arr[i]!=0 && arr[j]!=0 && abs(arr[i]-arr[j])<=1){
                    if (arr[i]<arr[j]){
                        arr[i]=0;
                    }
                    else if (arr[i]>arr[j]){
                        arr[j]=0;
                    }
                    else{
                        arr[i]=0;
                    }
                }
            }

        }

        cnt=0;
        for (int i = 0; i < n; i++){
            if (arr[i]!=0){
                cnt+=1;
            }
        }

        if (cnt==1){
            cout << "YES"<< endl;
        }else{
            cout << "NO"<< endl;
        }

    }
}

File: CFPractice/A_Remove_Smallest.py
tt = int(input())
for _ in range(tt):
    n = int(input())
    arr = list(map(int, input().split()))
    arr.sort()
    for i in range(n):
        for j in range(n):
            if i!=j and arr[i]!=0 and arr[j]!=0 and abs(arr[i]-arr[j])<=1:  
                if arr[i]<arr[j]:
                    arr[i]=0
                elif arr[i]>arr[j]:
                    arr[j]=0
                else:
                    arr[i]=0
    
    cnt=0
    for i in range(n):
        if arr[i]!=0:
            cnt+=1
    
    if cnt==1:
        print("YES")
    else:
        print("NO")


            

